src/Microsoft.OData.Client/ALinq/ExpressionWriter.cs

```diff
@@ -611,10 +611,18 @@ internal override Expression VisitUnary(UnaryExpression u)
                                 this.builder.Append(UriHelper.COMMA);
                             }
 
-                            this.builder.Append(UriHelper.QUOTE);
-                            this.builder.Append(UriHelper.GetTypeNameForUri(u.Type, this.context));
-                            this.builder.Append(UriHelper.QUOTE);
-                            this.builder.Append(UriHelper.RIGHTPAREN);
+                            if (this.builder.ToString().Contains(UriHelper.CAST))

```

Comment:
From line 606, it seems like we can't get here without `this.builder` containing "cast". If that is correct, then it seems like we are also missing the negative test that takes the `else` branch

Reply:
I'm not sure what you mean. To get to this line, 606 had to be called, which adds "cast" to the builder. Nothing that I can see between lines 606 and 614 will remove "cast" from the builder, so this `if` statement will always evaluate to `true` and we never take the else branch.

Reply:
yes I think the if--else is not necessary here.